packages/diagrams/backend/sirius-components-diagrams/src/main/java/org/eclipse/sirius/components/diagrams/components/NodeComponent.java

```diff
@@ -167,12 +168,19 @@ private Element doRender(VariableManager nodeVariableManager, String targetObjec
         LabelDescription labelDescription = nodeDescription.getLabelDescription();
         nodeVariableManager.put(LabelDescription.OWNER_ID, nodeId);
 
-        LabelType nodeLabelType = LabelType.INSIDE_CENTER;
-        if (containmentKind == NodeContainmentKind.BORDER_NODE) {
-            nodeLabelType = LabelType.OUTSIDE;
-        } else if (NodeType.NODE_IMAGE.equals(type)) {
-            nodeLabelType = LabelType.OUTSIDE_CENTER;
-        }
+        LabelType nodeLabelType = optionalPreviousNode.map(n -> LabelType.valueOfByValue(n.getLabel().getType())).orElseGet(() -> {
```

Comment:
Why are you using the previous node here? Unless I'm missing something, it will produce bugs. In Papaya, I have, on purpose, a node which, if it has no children is represented by an image and if it has some children, then it is represented as a rectangle. Thus between two rendering, the type of its label can change from inside to outside.

And if you don't use the previous node, you may not need the new method in `LabelType`

Reply:
Ok I didn't have this case in mind.
The problem here is we don't have access to the `SiriusWebLayoutConfigurator`. So we can't use the default values from this `SiriusWebLayoutConfigurator`.
That is the purpose of the `defaultLabelType` part. It may not correspond to the actual `SiriusWebLayoutConfigurator` default values.